src/test/test-fs-util.c

```diff
@@ -88,6 +90,34 @@ static void test_chase_symlinks(void) {
         assert_se(r >= 0);
         assert_se(path_equal(result, temp));
 
+        /* Paths that would "escape" outside of the "root" */
+
+        p = strjoina(temp, "/6dots");
+        assert_se(symlink("../../..", p) >= 0);
+
+        result = mfree(result);
+        r = chase_symlinks(p, temp, &result);
```

Comment:
Or to put it differently, we should not second-guess the kernel. If the kernel allows "../../foobar", so should we.